src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java

```diff
@@ -472,16 +463,30 @@ public static Object prepare(DataType type, Object value) {
         } else {
           return REFUSED_SENTINEL;
         }
+      case UNKNOWN:
+        if (value instanceof String[]) {
+          return Arrays.stream((String[]) value).collect(Collectors.joining(",", "{", "}"));
+        } else if (value instanceof String) {
+          return value;
+        } else {
+          return REFUSED_SENTINEL;
+        }
       default:
         Class<?> javaType = type.type;
         return javaType.isInstance(value) ? javaType.cast(value) : REFUSED_SENTINEL;
     }
   }
 
   private static Object defaultDecodeText(int len, ByteBuf buff) {
-    // Default to null
-    buff.skipBytes(len);
-    return null;
+    // decode unknown text values as text or as an array if it begins with `{`
```

Comment:
rather

```
    int idx = buff.readerIndex();
    if (idx < buff.writerIndex() && buff.getByte(idx) == '{') {
       ...
    }
```

this might be better since `readerIndex` is accessed once (`readableBytes()` returns `writerIndex - readerIndex`).